test/node/double_tests.js

```diff
@@ -38,26 +38,82 @@ describe('BSON Double Precision', function () {
+        { input: '-0.0', output: { $numberDouble: '-0.0' } },
+        { input: 3, output: { $numberDouble: '3.0' } },
+        { input: -3, output: { $numberDouble: '-3.0' } },
+        { input: 3.4, output: { $numberDouble: '3.4' } },
+        { input: Number.EPSILON, output: { $numberDouble: '2.220446049250313e-16' } },
+        { input: 12345e7, output: { $numberDouble: '123450000000.0' } },
+        { input: 12345e-1, output: { $numberDouble: '1234.5' } },
+        { input: -12345e-1, output: { $numberDouble: '-1234.5' } },
+        { input: Infinity, output: { $numberDouble: 'Infinity' } },
+        { input: -Infinity, output: { $numberDouble: '-Infinity' } },
+        { input: NaN, output: { $numberDouble: 'NaN' } },
+        {
+          input: Number.MAX_VALUE,
+          output: { $numberDouble: '1.7976931348623157e+308' }
+        },
+        { input: Number.MIN_VALUE, output: { $numberDouble: '5e-324' } },
+        {
+          input: -Number.MAX_VALUE,
+          output: { $numberDouble: '-1.7976931348623157e+308' }
+        },
+        { input: -Number.MIN_VALUE, output: { $numberDouble: '-5e-324' } },
+        // Reference: https://docs.oracle.com/cd/E19957-01/806-3568/ncg_math.html
+        {
+          // min positive normal number
+          input: '2.2250738585072014e-308',
+          output: { $numberDouble: '2.2250738585072014e-308' }
+        },
+        {
+          // max subnormal number (NOTE: JS does not output same input string, but numeric values are equal)
+          input: '2.225073858507201e-308',
+          output: { $numberDouble: '2.225073858507201e-308' }
+        },
+        {
+          // min positive subnormal number (NOTE: JS does not output same input string, but numeric values are equal)
+          input: '4.9406564584124654e-324',
+          output: { $numberDouble: '5e-324' }
+        },
+        {
+          // https://262.ecma-international.org/13.0/#sec-number.prototype.tofixed
+          // Note: calling toString on this integer returns 1000000000000000100, so toFixed is more precise
+          input: '1000000000000000128',
+          output: { $numberDouble: '1000000000000000128.0' }
+        }
       ];
 
       for (const test of tests) {
         const input = test.input;
         const output = test.output;
         const title = `returns ${inspect(output)} when Double is ${input}`;
         it(title, () => {
-          expect(output).to.deep.equal(input.toExtendedJSON({ relaxed: false }));
+          const inputAsDouble = new Double(input);
+          expect(inputAsDouble.toExtendedJSON({ relaxed: false })).to.deep.equal(output);
+          if (!Number.isNaN(inputAsDouble.value)) {
+            expect(Number(inputAsDouble.toExtendedJSON({ relaxed: false }).$numberDouble)).to.equal(
+              inputAsDouble.value
+            );
+          }
+        });
+
+        it(`input ${typeof input}: ${input} creates the same bytes after stringification`, () => {
```

Comment:
iiuc, the important aspect of this test is that the value of the double is preserved through ejson stringification.  we just happen to test this by asserting that the byte representations are equal.  Right?

If so, I propose we adjust the title to reflect what we're actually testing, and add a comment explaining that we determine that we're comparing bytes to assert that no loss of precision occurs.

```suggestion
        it(`preserves the precision of the number after stringification`, () => {
```

Reply:
Precision is related but more so that the string is acting as a correct data transfer format, we want to know that the same bytes can be reconstructed from the generated string, sometimes the string changes "4.9406564584124654e-324" -> "5e-324" but both represent the same ieee754 double bytes.